src/core/data/store.cc

```diff
@@ -123,12 +133,27 @@ FutureWrapper::FutureWrapper(
   assert(field_size > 0);
 #endif
   if (!read_only) {
-    auto mem_kind = find_memory_kind_for_executing_processor();
 #ifdef DEBUG_LEGATE
     assert(!initialize || future_.get_untyped_size() == field_size);
 #endif
-    auto p_init_value = initialize ? future_.get_buffer(mem_kind) : nullptr;
-    buffer_           = UntypedDeferredValue(field_size, mem_kind, p_init_value);
+    auto proc     = Processor::get_executing_processor();
+    auto mem_kind = proc.kind() == Processor::Kind::TOC_PROC ? Memory::Kind::GPU_FB_MEM
+                                                             : Memory::Kind::SYSTEM_MEM;
+    if (initialize) {
+      auto p_init_value = future_.get_buffer(mem_kind);
+#ifdef LEGATE_USE_CUDA
+      if (mem_kind == Memory::Kind::GPU_FB_MEM) {
+        // TODO: This should be done by Legion
```

Comment:
I guess `UntypedDeferredValue(field_size, mem_kind, p_init_value)` doesn't work if `mem_kind == Memory::Kind::GPU_FB_MEM`? Is this something we should report to be fixed in Legion?

Reply:
It doesn't work because `p_init_value` points to a framebuffer allocation whereas the runtime expects it to be a host allocation. We could potentially add another constructor that takes the initial value in a future and make the runtime issue a realm copy between the backing instances. @lightsighter what do you think?